### ISSUE TITLE
Make filtering of `Expect` request header configurable

### ISSUE THREAD

**Issue by jghiloni** [cloudfoundry/gorouter#121]
We have an application that simulates an S3 server. Requests to the application are signed, and the `Expect` header plays a role in that signature. I can see via commit 59e4d2c6606628ceb571064381c6be2930ed5e03 that the filtering of `Expect` headers has been there for at least 3 years, but is there any way we can opt-in to having that header propagated to applications?

**Comment by shashwathi** [comment]
Hi @jghiloni,
The `Expect` header is being removed by GoLang HTTP server.  A request is rejected with HTTP code 417 if any values other than `100-continue` is set to the header.

https://github.com/golang/go/blob/release-branch.go1.5/src/net/http/server.go#L1344-L1354

However, GoLang 1.6 allows `Expect: 100-continue` to propagate.
https://github.com/golang/go/blob/release-branch.go1.6/src/net/http/server.go#L1455-L1465

We're planning to upgrade to GoLang 1.6 in the near future.

Regards
@leochu and @shashwathi
CF Routing Team

**Comment by jghiloni** [comment]
Thanks @shashwathi! Are there stories in Pivotal Tracker we can track this with?

**Comment by shashwathi** [comment]
@shalako Will have better idea about this.


Regards
@atulkc and @shashwathi

**Comment by shalako** [comment]
@jghiloni Hello, you'll see in the second comment of this issue the link to the Tracker story that was automatically created. We'll keep this story around until we upgrade to golang 1.6, then use it to verify the fix.

**Comment by jghiloni** [comment]
Perfect, thanks!

**Comment by shalako** [comment]
Hello @jghiloni,

We have verified that after upgrading gorouter to golang 1.6, the `Expect` header is now persisted. 

However, Amazon ELB does not persist this header so requests to Gorouter that first transit an ELB will not include the `Expect` header. For the same reason, if you're leveraging a route service that runs behind an ELB, the header will be stripped if requests for a route in CF transit that route service.

Closing this issue.
